pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -336,6 +356,36 @@ func (c *Controller) Cluster() string {
 	return c.clusterID
 }
 
+// IstiodEvent is an event associated with a Istiod tenant. In K8S namespace is the tenancy unit - we may
+// not run deployments or services in the cluster (for example central istiod), but a namespace is expected.
+func (c *Controller) IstiodEvent(ns, reason, msg string, warn bool) {
+	if c.Recoder == nil {
+		return
+	}
+	// Event type can be normal and warning
+	evType := v1.EventTypeNormal
+	if warn {
+		evType = v1.EventTypeWarning
+	}
+	// reference is required
+	p, err := c.nsLister.Get(ns)
```

Comment:
I don't think we want the nslister here. All its used for is for the log, which I don't think is a useful log anyways?

Reply:
Right.